src/components/EventUserTable.js

```diff
@@ -1,54 +1,300 @@
+        const yearInt = parseInt(user.year);
+        if (yearInt) {
+          years[yearInt] = years[yearInt] ? years[yearInt] + 1 : 1;
+        }
+      }
+      if (user.diet) {
+        dietary[user.diet] = dietary[user.diet] ? dietary[user.diet] + 1 : 1;
+      }
+      if (user.gender) {
+        genders[user.gender] = genders[user.gender] ? genders[user.gender] + 1 : 1;
+      }
+    })
+
+    this.setState({
+      faculties,
+      years,
+      genders,
+      dietary
+    });
+  }
+
+  async updateEventTableData(eventID) {
+    const eventParams = new URLSearchParams({
+      id: eventID
+    });
+
+    fetchBackend('/events/getUsers?' + eventParams, 'GET')
+      .then(response => response.json())
+      .then(async users => {
+        this.registrationNumbers(users)
       });
   }
 
   componentDidMount() {
     this.getEventTableData(this.props.event.id);
   }
 
+  /*
+    the if statement is only used if an exec goes from one event directly to another event (not implemented right now)
+  */
   componentDidUpdate(prevProps) {
     if (prevProps.event.id !== this.props.event.id) {
-      this.getEventTableData(this.props.event.id);
+      this.updateEventTableData(this.props.event.id);
+    }
+  }
+
+  /*
+    changes the visibility of the stats of whichever div was selected
+  */
+  changeVisibility(event) {
+    const id = event.target.id
+    const invisible = { visible: false, style: { display: 'none' } }
+    const visible = { visible: true, style: { display: 'flex', paddingBottom: '20px', paddingLeft: '50px' } }
+    switch (id) {
+      case 'registration':
+        if (this.state.registrationVisible.visible) {
+          this.setState({ registrationVisible: invisible })
+        } else {
+          this.setState({ registrationVisible: visible })
+        }
+        break;
+      case 'faculties':
+        if (this.state.facultyVisible.visible) {
+          this.setState({ facultyVisible: invisible })
+        } else {
+          this.setState({ facultyVisible: visible })
+        }
+        break;
+      case 'year':
+        if (this.state.yearVisible.visible) {
+          this.setState({ yearVisible: invisible })
+        } else {
+          this.setState({ yearVisible: visible })
+        }
+        break;
+      case 'dietary':
+        if (this.state.dietaryVisible.visible) {
+          this.setState({ dietaryVisible: invisible })
+        } else {
+          this.setState({ dietaryVisible: visible })
+        }
+        break;
+      case 'gender':
+        if (this.state.gendersVisible.visible) {
+          this.setState({ gendersVisible: invisible })
+        } else {
+          this.setState({ gendersVisible: visible })
+        }
+        break;
+      case 'heardFrom':
+        if (this.state.heardFromVisible.visible) {
+          this.setState({ heardFromVisible: invisible })
+        } else {
+          this.setState({ heardFromVisible: visible })
+        }
+        break;
+      default:
```

Comment:
I think this might work 

```
if (this.state[id].visible) {
  this.setState({[id]: invisible})
} else {
  this.setState({[id]: visible})
}
```

Reply:
bless, i was trynna do this.setState({ someName: invisible}) which didnt work and never tried the square brackets